extensions/wikia/Wall/index/CommentsIndex.class.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+/**
+ * CommentsIndex Class
+ * @author Kyle Florence, Saipetch Kongkatong, Tomasz Odrobny
+ */
+
+class CommentsIndex {
+	use \Wikia\Logger\Loggable;
+
+	/** @var CommentsIndex $instance */
+	private static $instance = null;
+
+	/** @var CommentsIndexEntry[] $objectCache */
+	private $objectCache = [];
+
+	private function __construct() {
+	}
+
+	public static function singleton(): CommentsIndex {
```

Comment:
can we rename this method to `instance` or `getInstance`? AFAIR it is more common in this repo